api/user_test.go

```diff
@@ -201,3 +202,57 @@ func Test_CvpGetUserValid_UnitTest(t *testing.T) {
 	}
 
 }
+func Test_CvpAddNilUser_UnitTest(t *testing.T) {
+	nilUserErr := errors.New("can not add nil user")
+	client := NewMockClient("", nilUserErr)
+	api := NewCvpRestAPI(client)
+	expectedErr := errors.Errorf("AddUser: %s", nilUserErr.Error())
+
+	if err := api.AddUser(nil); err.Error() != expectedErr.Error() {
+		t.Fatalf("Expected error: [%v] But received: [%v]", expectedErr, err)
+	}
+}
+
+func Test_CvpAddUserAlreadyExists_UnitTest(t *testing.T) {
```

Comment:
Why not use table driven tests here too? That is easier to maintain and add new tests.

Reply:
+1 on table test

Reply:
By table test, you mean put test cases as subtests in 1 single AddUser test, right ?

Reply:
This could be done in later commits. I have raised an issue regarding this.